Framework/PythonInterface/plugins/algorithms/SaveISISReflectometryORSO.py

```diff
@@ -141,15 +151,22 @@ def _set_name(self):
         if self.is_stitched:
             self._name = "Stitched"
         elif self._q_conversion_theta is not None:
-            self._name = str(self._q_conversion_theta)
+            self._name = f"{self._q_conversion_theta:.3f}"
         else:
             self._name = self._ws.name()
 
+        if self.is_polarized:
+            self._name = f"{self._name} {self._spin_state}"
+            return
+
```

Comment:
```suggestion
        if self.is_polarized:
            self._name = f"{self._name} {self._spin_state}"
            return

```
Here I have decided to return early because if the dataset is polarized is not necessary to prepend the ws name for unique datasets and in some cases the ws name would be repeated.